### PR TITLE
1953 endpoint delete prescription & prescription line

### LINKED ISSUE #1953: Endpoint: Delete Prescription & Prescription Line
## Describe the solution you'd like 🎁
Delete invoice 

```graphql
DeletePrescription {
    id: String
}
```

```graphql
DeletePrescriptionError {
   InvoiceDoesNotExist
   NotThisStoreInvoice
   CannotEditFinalised
   NotAPrescriptionInvoice
   DatabaseError(RepositoryError)
}
```

Delete invoice line
```graphql
DeletePrescriptionLine {
    id: String
}
```

```graphql
DeletePrescriptionLineError {
   LineDoesNotExist
   InvoiceDoesNotExist
   NotThisStoreInvoice
   CannotEditFinalised
   NotAPrescriptionInvoice
   DatabaseError(RepositoryError)
}
```

### Moneyworks Jobcode 🧰
OMS:MVP | Open mSupply MVP Prescription Module
-- | --

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #1953 

# 👩🏻‍💻 What does this PR do? 
Implement delete invoice and invoice line for prescriptions

# 🧪 How has/should this change been tested? 
Run tests
- Make sure functionality of existing Outbound Shipment deletes still work.

### CODE REVIEW

File: server/service/src/invoice/prescription/delete/mod.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+pub enum DeletePrescriptionError {
+    InvoiceDoesNotExist,
+    DatabaseError(RepositoryError),
+    NotThisStoreInvoice,
+    CannotEditFinalised,
+    LineDeleteError {
+        line_id: String,
+        error: DeleteStockOutLineError,
+    },
+    NotAPrescription,
```
Comment (mark-prins) [inline server/service/src/invoice/prescription/delete/mod.rs:70]: A query: In the issue this is `NotAPrescriptionInvoice`. I like `NotAPrescription` but it is inconsistent with `NotThisStoreInovice`

File: server/service/src/invoice/prescription/delete/mod.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+#[derive(Debug, PartialEq, Clone)]
+
+pub enum DeletePrescriptionError {
+    InvoiceDoesNotExist,
+    DatabaseError(RepositoryError),
+    NotThisStoreInvoice,
+    CannotEditFinalised,
+    LineDeleteError {
```
Comment (mark-prins) [inline server/service/src/invoice/prescription/delete/mod.rs:66]: this isn't in the issue - it's an addition?

**Review by mark-prins — APPROVED** [review]
👍  working nicely for me

File: server/service/src/invoice/prescription/delete/mod.rs (hunk 2 above)
Comment (roxy-dao) [reply server/service/src/invoice/prescription/delete/mod.rs:66]: Yes (:

**Pull request merged**